sql/core/src/main/scala/org/apache/spark/sql/execution/command/UnsetNamespacePropertiesCommand.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command
+
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.catalyst.analysis.ResolvedNamespace
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.CatalogHelper
+import org.apache.spark.sql.connector.catalog.NamespaceChange
+import org.apache.spark.sql.errors.QueryCompilationErrors
+
+/**
+ * A command that ALTER NAMESPACE UNSET PROPERTIES command.
+ *
+ * The syntax of this command is:
+ * {{{
+ *     ALTER (DATABASE|SCHEMA|NAMESPACE) ... UNSET (DBPROPERTIES|PROPERTIES) [IF EXISTS] ...;
+ * }}}
+ */
+case class UnsetNamespacePropertiesCommand(
+    ident: LogicalPlan,
+    propKeys: Seq[String],
+    ifExists: Boolean) extends UnaryRunnableCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val ResolvedNamespace(catalog, ns, metadata) = child
+    if (!ifExists) {
+      val nonexistentKeys = propKeys.filter(key => !metadata.contains(key))
+      if (nonexistentKeys.nonEmpty) {
+        throw QueryCompilationErrors.unsetNonExistentPropertiesError(
+          nonexistentKeys, ns)
+      }
+    }
+    val changes = propKeys.map {
+      NamespaceChange.removeProperty
+    }
+    catalog.asNamespaceCatalog.alterNamespace(ns.toArray, changes: _*)
```

Comment:
can we check the API doc of `alterNamespace`? Will it throw error if the to-be-removed property does not exist?

Reply:
From the API doc of `alterNamespace`, it can be seen that when deleting non-existent property, the API does not throw `any exceptions` or `prompts`, resulting in silence:
https://github.com/apache/spark/blob/b49479b43e1f95ba0588ed6a53f410969c0785f4/sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsNamespaces.java#L127-L137
https://github.com/apache/spark/blob/b49479b43e1f95ba0588ed6a53f410969c0785f4/sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala#L96-L97

Reply:
Then why do we add the `IF EXISTS` clause... I feel it just introduces troubles with no benefits. Shall we not have it and simply say non-existing property will be ignored when UNSET?

Reply:
Because this is a new `sql command`, I think we can skip the `IF EXISTS` clause
but I would like to give a `counter-example`, for the `table`, we have the following existed sql command:
```
ALTER (TABLE | VIEW) ... SET TBLPROPERTIES ...
ALTER (TABLE | VIEW) ... UNSET TBLPROPERTIES (IF EXISTS)? ...
```
https://github.com/apache/spark/blob/4cf54508fd23ed5acd45a2d428241f080ac5bc9c/sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4#L148-L151
Should we align with it?

Reply:
We can't change the table syntax now as it will be a breaking change (parser error), but we can simply ignore the IF EXISTS clause and also remove it from document.

Reply:
Okay, Let me first remove `IF EXISTS` from this PR implementation.
Also, I will update the related doc in another PR `Unify v1 and v2 ALTER TABLE .. UNSET TBLPROPERTIES tests` at the same time.

Reply:
> Then why do we add the `IF EXISTS` clause... I feel it just introduces troubles with no benefits. Shall we not have it and simply say non-existing property will be ignored when UNSET?

Updated.

Reply:
I also updated the document `docs/sql-ref-syntax-ddl-alter-database.md`